ElfUtils/ElfFile.cpp

```diff
@@ -253,26 +253,26 @@ bool ElfFileImpl<llvm::object::ELF32LE>::Is64Bit() const {
 
 }  // namespace
 
-ErrorMessageOr<std::unique_ptr<ElfFile>> ElfFile::CreateFromBuffer(std::string_view file_path,
-                                                                   const void* buf, size_t len) {
+ErrorMessageOr<std::unique_ptr<ElfFile>> ElfFile::CreateFromBuffer(
+    const std::filesystem::path& file_path, const void* buf, size_t len) {
   std::unique_ptr<llvm::MemoryBuffer> buffer = llvm::MemoryBuffer::getMemBuffer(
       llvm::StringRef(static_cast<const char*>(buf), len), llvm::StringRef("buffer name"), false);
   llvm::Expected<std::unique_ptr<llvm::object::ObjectFile>> object_file_or_error =
       llvm::object::ObjectFile::createObjectFile(buffer->getMemBufferRef());
 
   if (!object_file_or_error) {
-    return ErrorMessage(absl::StrFormat("Unable to load ELF file \"%s\": %s", file_path,
+    return ErrorMessage(absl::StrFormat("Unable to load ELF file \"%s\": %s", file_path.string(),
                                         llvm::toString(object_file_or_error.takeError())));
   }
 
   return ElfFile::Create(file_path, llvm::object::OwningBinary<llvm::object::ObjectFile>(
                                         std::move(object_file_or_error.get()), std::move(buffer)));
 }
 
-ErrorMessageOr<std::unique_ptr<ElfFile>> ElfFile::Create(std::string_view file_path) {
+ErrorMessageOr<std::unique_ptr<ElfFile>> ElfFile::Create(const std::filesystem::path& file_path) {
   // TODO(hebecker): Remove this explicit construction of StringRef when we
   // switch to LLVM10.
-  const llvm::StringRef file_path_llvm{file_path.data(), file_path.size()};
+  const llvm::StringRef file_path_llvm{file_path.c_str()};
```

Comment:
Personal preference: I'd say `file_path.native()` would be a tiny bit nicer than `file_path.c_str()` because that superfluous call to strlen under the hood hurts. But I don't feel particularly strong about that. :-)

Reply:
this apparently requires a string intermediate variable on windows so I introduced one. which means this c_str is now called on a string.